third_party/validator/compute_instance.go

```diff
@@ -0,0 +1,304 @@
+		return nil, err
+	}
+	log.Printf("[DEBUG] Loading zone: %s", z)
+	zone, err := config.clientCompute.Zones.Get(
+		project, z).Do()
+	if err != nil {
+		return nil, fmt.Errorf("Error loading zone '%s': %s", z, err)
+	}
+
+	instance, err := expandComputeInstance(project, zone, d, config)
+	if err != nil {
+		return nil, err
+	}
+
+	return jsonMap(instance)
+}
+
+func expandComputeInstance(project string, zone *compute.Zone, d TerraformResourceData, config *Config) (*computeBeta.Instance, error) {
+	// Get the machine type
+	var machineTypeUrl string
+	if mt, ok := d.GetOk("machine_type"); ok {
+		log.Printf("[DEBUG] Loading machine type: %s", mt.(string))
+		machineType, err := config.clientCompute.MachineTypes.Get(
+			project, zone.Name, mt.(string)).Do()
+		if err != nil {
+			return nil, fmt.Errorf(
+				"Error loading machine type: %s",
+				err)
+		}
+		machineTypeUrl = machineType.SelfLink
+	}
+
+	// Build up the list of disks
+
+	disks := []*computeBeta.AttachedDisk{}
+	if _, hasBootDisk := d.GetOk("boot_disk"); hasBootDisk {
+		bootDisk, err := expandBootDisk(d, config, zone, project)
+		if err != nil {
+			return nil, err
+		}
+		disks = append(disks, bootDisk)
+	}
+
+	if _, hasScratchDisk := d.GetOk("scratch_disk"); hasScratchDisk {
+		scratchDisks, err := expandScratchDisks(d, config, zone, project)
+		if err != nil {
+			return nil, err
+		}
+		disks = append(disks, scratchDisks...)
+	}
+
+	attachedDisksCount := d.Get("attached_disk.#").(int)
+
+	for i := 0; i < attachedDisksCount; i++ {
+		diskConfig := d.Get(fmt.Sprintf("attached_disk.%d", i)).(map[string]interface{})
+		disk, err := expandAttachedDisk(diskConfig, d, config)
+		if err != nil {
+			return nil, err
+		}
+
+		disks = append(disks, disk)
+	}
+
+	sch := d.Get("scheduling").([]interface{})
+	var scheduling *computeBeta.Scheduling
+	if len(sch) == 0 {
+		// TF doesn't do anything about defaults inside of nested objects, so if
+		// scheduling hasn't been set, then send it with its default values.
+		scheduling = &computeBeta.Scheduling{
+			AutomaticRestart: googleapi.Bool(true),
+		}
+	} else {
+		prefix := "scheduling.0"
+		scheduling = &computeBeta.Scheduling{
+			AutomaticRestart:  googleapi.Bool(d.Get(prefix + ".automatic_restart").(bool)),
+			Preemptible:       d.Get(prefix + ".preemptible").(bool),
+			OnHostMaintenance: d.Get(prefix + ".on_host_maintenance").(string),
+			ForceSendFields:   []string{"AutomaticRestart", "Preemptible"},
+		}
+	}
+
+	metadata, err := resourceInstanceMetadata(d)
+	if err != nil {
+		return nil, fmt.Errorf("Error creating metadata: %s", err)
+	}
+
+	networkInterfaces, err := expandNetworkInterfaces(d, config)
+	if err != nil {
+		return nil, fmt.Errorf("Error creating network interfaces: %s", err)
+	}
+
+	accels, err := expandInstanceGuestAccelerators(d, config)
+	if err != nil {
+		return nil, fmt.Errorf("Error creating guest accelerators: %s", err)
+	}
+
+	// Create the instance information
+	return &computeBeta.Instance{
```

Comment:
We use the computeBeta library for legacy reasons. If you find that this becomes a problem you can probably move to the GA version.